src/main/scala/com/fulcrumgenomics/bam/IdentifyPrimers.scala

```diff
@@ -0,0 +1,1027 @@
+
+  import IdentifyPrimers._
+
+  private val kswExecutable: Option[FilePath] = this.ksw.map(Paths.get(_)).flatMap {
+    case p if Files.exists(p) => Some(p)
+    case name =>
+      val path = System.getenv("PATH")
+      validate(path != null, "PATH environment variable was not found; required for --ksw")
+      val systemPath = path.split(File.pathSeparatorChar).view.map(PathUtil.pathTo(_))
+      systemPath.map(p => p.resolve(name)).find(ex => Files.exists(ex))
+        .orElse {
+          throw new ValidationException(s"Could not find ksw executable ${ksw} in PATH: $path")
+        }
+  }
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+  Io.assertCanWriteFile(metrics)
+  kswExecutable.foreach(Io.assertReadable)
+
+  validate(maxMismatches >= 0,   "--max-mismatches must be >= 0")
+  validate(minMismatchDelta > 0, "--min-mismatch-delta must be > 0")
+  validate(matchScore > 0,       "--match-score must be > 0")
+  validate(mismatchScore < 0,    "--mismatch-score must be < 0")
+  validate(gapOpen < 0,          "--gap-open must be < 0")
+  validate(gapExtend < 0,        "--gap-extend must be < 0")
+  maxTemplatesInRam.foreach { m => validate(templatesPerThread < m, "--max-templates-in-ram must be greater than or equal to --templates-per-thread")}
+
+  private val aligner: Aligner = Aligner(matchScore = matchScore, mismatchScore = mismatchScore, gapOpen = gapOpen, gapExtend = gapExtend, mode = AlignmentMode.Glocal)
+
+  private val (fwdMatcher, revMatcher) = PrimerMatcher.build(
+    path              = this.primers,
+    aligner           = aligner,
+    slop              = slop,
+    maxMismatches     = maxMismatches,
+    minMismatchDelta  = minMismatchDelta,
+    minAlignmentScore = minAlignmentScore,
+    withFullAlignment = !skipFullAlignment,
+    revComp           = true
+  )
+
+  var numAlignments: AtomicLong = new AtomicLong(0)
+
+  private val PrimerPairMatchTypeTag: String = "pp"
+  private val PrimerInfoForwardTag: String   = "pf"
+  private val PrimerInfoReverseTag: String   = "pr"
+  private val NoPrimerMatchInfo: String      = "none"
+  private val comments: Seq[String]          = {
+    Seq(
+      "The pf/pr tags store the primer match metadata for the forward and reverse strand respectively.",
+      "The pf/pr tags are formatted as follow: <pair_id>,<primer_id>,<refName>:<start>-<end>,<strand>,<5-prime-offset>,<match-type>,<match-type-info>.",
+      s"The match-type is 'location', 'mismatch', or 'aligmnent' based on if the match was found using the location, mismatch-based alignment, or full-alignment.",
+      s"The pp tag is either 'canonical', 'non-canonical', 'single', or '$NoPrimerMatchInfo' based on if the matches for the primer pair agree, disagree, or are missing."
+    )
+  }
+
+  override def execute(): Unit = {
+    this.kswExecutable match {
+      case Some(p) => logger.info(s"Using ksw aligner: $p")
+      case None    => logger.info("Using the scala aligner.")
```

Comment:
How about `logger.info("Using the internal aligner; install ksw for faster operation.")`